tests/playwright/src/specs/kubernetes-context-smoke.spec.ts

```diff
@@ -0,0 +1,125 @@
+  });
+
+  test('Can load kube contexts in Kubernetes page', async ({ navigationBar }) => {
+    const settingsBar = await navigationBar.openSettings();
+    const kubePage = await settingsBar.openTabPage(KubeContextPage);
+    await playExpect(kubePage.heading).toBeVisible();
+
+    await playExpect.poll(async () => await kubePage.pageIsEmpty(), { timeout: 10000 }).toBeFalsy();
+    for (const context of testContexts) {
+      const row = await kubePage.getContextRowByName(context);
+      if (row === undefined) {
+        throw Error(`Context: '${context}' not found`);
+      }
+      await playExpect(row.getByLabel('Context Name', { exact: true })).toHaveText(context);
+      await playExpect(row.getByLabel('Context Cluster')).toHaveText(context + '-cluster');
+      await playExpect(row.getByLabel('Context Server')).toHaveText(context + '-server');
+      await playExpect(row.getByLabel('Context User')).toHaveText(context + '-user');
+
+      if (context === 'context-1') {
+        // check if the context is default
+        await playExpect.poll(async () => await kubePage.isContextDefault(context), { timeout: 10000 }).toBeTruthy();
+        await playExpect(row.getByLabel('Set as Current Context')).not.toBeVisible();
+        await playExpect(row.getByLabel('Context Namespace')).toHaveText(context + '-namespace');
+      } else {
+        await playExpect(row.getByLabel('Set as Current Context')).toBeVisible();
+      }
+    }
+  });
+
+  test('Can switch default context', async ({ navigationBar }) => {
+    const settingsBar = await navigationBar.openSettings();
+    const kubePage = await settingsBar.openTabPage(KubeContextPage);
+    await playExpect(kubePage.heading).toBeVisible();
+
+    await kubePage.setDefaultContext('context-2');
+    // check that switch worked - current context banner should be visible
+    playExpect(await kubePage.isContextDefault('context-1')).toBeFalsy();
+    await playExpect.poll(async () => await kubePage.isContextDefault('context-2'), { timeout: 10000 }).toBeTruthy();
+    playExpect(await kubePage.isContextDefault('context-3')).toBeFalsy();
+  });
+
+  test('Can delete all contexts from Kubernetes Contexts page', async ({ navigationBar }) => {
+    const settingsBar = await navigationBar.openSettings();
+    const kubePage = await settingsBar.openTabPage(KubeContextPage);
+    await playExpect(kubePage.heading).toBeVisible();
+
+    for (const context of testContexts) {
+      // confirmation only pops up if the context is default
+      if (await kubePage.isContextDefault(context)) {
+        await kubePage.deleteContext(context);
+      } else {
+        await kubePage.deleteContext(context, false);
+      }
+      await playExpect(kubePage.heading).toBeVisible();
+    }
+
+    for (const context of testContexts) {
+      await playExpect
+        .poll(async () => await kubePage.getContextRowByName(context), { timeout: 10000 })
+        .toBeUndefined();
```

Comment:
This approach is vulnerable to a race condition, since getContextRowByName() only returns undefined when the page is empty, however deletion is an async process as such it's possible that deleting the last row will not have been completed before the validation on the first context is done in this check, which would cause it to throw.

```suggestion
      await playExpect(kubePage.getContextRowByName(context)).not.toBeVisible();
```